contrib/internal/httptrace/config.go

```diff
@@ -0,0 +1,52 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package httptrace
+
+import (
+	"os"
+	"regexp"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+)
+
+const (
+	// envQueryStringDisabled is the name of the env var used to disabled query string collection.
+	envQueryStringDisabled = "DD_TRACE_HTTP_URL_QUERY_STRING_DISABLED"
+	// envQueryStringRegexp is the name of the env var used to specify the regexp to use for query string obfuscation.
+	envQueryStringRegexp = "DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP"
+	// envClientIPHeader is the name of the env var used to specify the IP header to be used for client IP collection.
+	envClientIPHeader = "DD_TRACE_CLIENT_IP_HEADER"
+	// envClientIPHeader is the name of the env var used to disable client IP tag collection.
+	envClientIPHeaderDisabled = "DD_TRACE_CLIENT_IP_HEADER_DISABLED"
+)
+
+// defaultQueryStringRegexp is the regexp used for query string obfuscation if `envQueryStringRegexp` is empty.
+// The regexp is taken from https://datadoghq.atlassian.net/wiki/spaces/APS/pages/2490990623/QueryString+-+Sensitive+Data+Obfuscation
+var defaultQueryStringRegexp = regexp.MustCompile("(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|token|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)(?:(?:\\s|%20)*(?:=|%3D)[^&]+|(?:\"|%22)(?:\\s|%20)*(?::|%3A)(?:\\s|%20)*(?:\"|%22)(?:%2[^2]|%[^2]|[^\"%])+(?:\"|%22))|bearer(?:\\s|%20)+[a-z0-9\\._\\-]|token(?::|%3A)[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L](?:[\\w=-]|%3D)+\\.ey[I-L](?:[\\w=-]|%3D)+(?:\\.(?:[\\w.+\\/=-]|%3D|%2F|%2B)+)?|[\\-]{5}BEGIN(?:[a-z\\s]|%20)+PRIVATE(?:\\s|%20)KEY[\\-]{5}[^\\-]+[\\-]{5}END(?:[a-z\\s]|%20)+PRIVATE(?:\\s|%20)KEY|ssh-rsa(?:\\s|%20)*(?:[a-z0-9\\/\\.+]|%2F|%5C|%2B){100,}")
+
+type config struct {
+	queryStringRegexp *regexp.Regexp
+	clientIPHeader    string
+	clientIP          bool
+	queryString       bool
+}
+
+func newConfig() config {
+	c := config{
+		clientIPHeader:    os.Getenv(envClientIPHeader),
+		clientIP:          !internal.BoolEnv(envClientIPHeaderDisabled, false),
+		queryString:       !internal.BoolEnv(envQueryStringDisabled, false),
+		queryStringRegexp: defaultQueryStringRegexp,
+	}
+	if s, set := os.LookupEnv(envQueryStringRegexp); !set {
+		return c
+	} else if s == "" {
+		c.queryStringRegexp = nil
```

Comment:
Might be worth logging a debug level message here, that query string obfuscation is disabled because of empty env. var. I'm thinking it may help in some support ticket if someone is ticking they will be resetting the defaults by setting that env. var to empty, when instead setting it to empty will disable it. Isn't that confusing?

Reply:
Agreed, a log message will be helpful. As for the intended use of the regexp, I don't think there should be too much confusion since the usage will be documented (see [here](https://docs-staging.datadoghq.com/robert/additional-data-collection/tracing/setup_overview/configure_data_security/?tab=mongodb#redacting-the-query-in-the-url)), and a link to this documentation is provided in the release notes. I can add a link to this in the comments of `config.go` as well.
This also helps in cases where there is a misconfig client side and something like `DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP=$VAR` where `$VAR` is actually empty happens. And probably some other cases.

Reply:
See 5b5229e9bc4ca25d2abbf84ab77eec036fe26f36